<a href="https://colab.research.google.com/github/min02yam/ADP/blob/main/25_statistical_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

통계분석 

# 1.평균속도 - 조화평균
서울에서 영동까지 100km/h로 이동하였으며, 그 후 영동에서 서울까지 80km/h의 속도로 되돌아왔을 때, 평균속도를 계산하시오.

In [ ]:
a = 100
b = 80
cat("정답은",round((2*a*b)/(a+b),2))

정답은 88.89

# 2.연평균 생산량 증가율 - 기하평균
어느 공장의 작년 생산량이 1000개, 올해 생산량이 2000개, 내년 생산량이 3000개로 늘어날 예정이다. 연평균 생산량 증가율을 계산하시오.

In [ ]:
n=2
a= 1000
b= 3000
ans = sqrt(b/a)-1  
cat("연평균 증가율은", round(ans*100,2) ,"%","입니다." )

연평균 증가율은 73.21 % 입니다.

# 3.조건부 확률
등산을 좋아하는 사람 200명 중 남자 130명, 여자 70명이 있다. 등산을 좋아하지않는사람 100명 중 남자 30명, 여자 70명이 있다. 남자일때 등산을 좋아할 확률을 구하시오.


In [ ]:
cat("확률은" , round( 130/(130+30), 2),"입니다." )

확률은 0.81 입니다.

# 4.표본 분산의 신뢰구간
표본 개수 20개의 표본분산이 90^2일때, 신뢰도 95%에서 표본분산의 신뢰구간을 구하시오.

In [ ]:
df=19
s=90
L = df * s / qchisq(0.975,df)
U = df* s / qchisq(0.025,df)

In [ ]:
cat("95% 신뢰구간은", "[",round(L,2),",",round(U,2),"]" ,"입니다.")

95% 신뢰구간은 [ 52.05 , 191.99 ] 입니다.

# 5.짝표본 모평균 차 검정
혈압약의 효과를 검증하려고 한다. 임상 대상 20명에 대하여 혈압약을 먹었을때 혈압차이가 24mmHg, 표준편차가 9.5mmHg로 집계되었다.

In [ ]:
n=20
d=24
sigma=9.5
t=d/(sigma/sqrt(n))

In [ ]:
paste("t검정통계량의 값은",round(t,4) )
paste("유의확률은", 1-pt(t, n-1) )

[1] "t검정통계량의 값은 11.298"

[1] "유의확률은 3.56065954498774e-10"

- `H_0`: 치료 전후의 혈압의 평균값은 차이가 없다.
- `H_a`: 치료 후 혈압의 평균값이 더 낮다.
- `t검정통계량값`: 11.298
- `결론`: t검정통계량의 값이 11.298일때 유의확률은 0과 가까울 정도로 매우 작은 수치로 유의수준 0.05보다 작다.<br> 그러므로 유의수준 0.05하에서는 통계적으로 귀무가설을 기각할 근거가 충분하다. <br>즉, 귀무가설을 기각한다. 
- `해석`: 유의수준 0.05하에서는 혈압약의 효과가 있다고 할 수 있다. 

# 6.크루스칼왈리스-중심위치

In [ ]:
A = c(4,4,2,3,4)
B = c(6,4,4,5,6)
C = c(6,6,7,6,5)
x = c(A, B, C)
g = factor(rep(1:3, c(5, 5, 5)),
            labels = c("공장A",
                       "공장B",
                       "공장C"))
kruskal.test(x, g)


	Kruskal-Wallis rank sum test

data:  x and g
Kruskal-Wallis chi-squared = 9.3927, df = 2, p-value = 0.009129


- `귀무가설(H_0)`: 세 공장 A, B, C 의 생산품 무게의 중앙값은 같다.
- `연구가설(H_a)`: 세 공장 A, B, C의 생산품 무게의 중앙값은 같지 않다.
- `카이스퀘어 검정통계량의 값` : 9.3927
- `결론`: 카이스퀘어 검정통계량의 값이 9.3927일때 유의확률은 0.0091로써 유의수준 0.05보다 작다. 그러므로 유의수준 0.05하에서는 통계적으로 귀무가설을 기각할 근거가 충분하다.
- `해석`: 즉, 유의수준 0.05하에서는 세 공장 A,B,C,간 생산량의 차이가 있다고 할 수 있다. 

# 7.최적투자안 ,,, 

In [ ]:
<공부중>